mmcv/runner/hooks/optimizer.py

```diff
@@ -38,7 +44,10 @@ def after_train_iter(self, runner):
 class Fp16OptimizerHook(OptimizerHook):
```

Comment:
Since the logic of torch.amp and our own implementation is quite different, we may implement two classes.
```python
if TORCH_VERSION != 'parrots' and TORCH_VERSION >= '1.6.0':
    class Fp16OptimizerHook:
        # amp implementation
else:
    class Fp16OptimizerHook:
        # our own implementation
```
It should be clearer for users.